flipt-client-node/README.md

```diff
@@ -66,27 +62,75 @@ The `FliptEvaluationClient` supports the following authentication strategies:
 - [Client Token Authentication](https://docs.flipt.io/authentication/using-tokens)
 - [JWT Authentication](https://docs.flipt.io/authentication/using-jwts)
 
-## Memory Management
+### Custom Fetcher
 
-Since TypeScript/JavaScript is a garbage collected language there is no concept of "freeing" memory. We have to allocate memory for the engine through the `initialize_engine` FFI call.
+The `FliptEvaluationClient` supports custom fetchers. This allows you to fetch flag state from a custom source or override HTTP headers.
 
-Make sure to call the `close` method on the `FliptEvaluationClient` class once you are done using it.
+The fetcher can be passed in as an argument to the `FliptEvaluationClient` initializer function.
+
+```typescript
+const fliptEvaluationClient = await FliptEvaluationClient.init('default', {
+  url: 'http://localhost:8080',
+  authentication: {
+    clientToken
+  },
+  fetcher: customFetcher
+});
+```
+
+The fetcher is a function that takes an optional [`IFetcherOpts`](https://github.com/flipt-io/flipt-client-sdks/blob/4821cb227c6c8b10419b96674d44ad1d6668a647/flipt-client-browser/src/models.ts#L1) argument and returns a `Promise` that resolves to a `Response` object.
+
+## Timer Management
+
+The `FliptEvaluationClient` class uses a timer to fetch new flag state at a regular interval.
+
+Make sure to call the `close` method on the `FliptEvaluationClient` class once you are done using it to stop the timer and clean up resources.
 
 ```typescript
 fliptEvaluationClient.close();
 ```
 
-## Potential Issues
+## State Management
 
-### Vitest
+The `FliptEvaluationClient` class pulls flag state from the Flipt instance at the `url` provided in the `options` object on instantiation.
 
-If you are using this library in combination with [vitest](https://github.com/vitest-dev/vitest), you may run into segmentation fault issues. This is due to the fact that `vitest` does not work nicely with `ffi-napi` by default.
+This state is pulled from the Flipt instance on instantiation and every `update_interval` seconds thereafter.
 
-See the following issue for more information: [vitest #2091: vitest will crash when using ffi-napi](https://github.com/vitest-dev/vitest/issues/2091).
+To update the flag state manually, you can call the `refresh` method on the `FliptEvaluationClient` class.
+
+```typescript
+// Refresh the flag state
+fliptEvaluationClient.refresh();
+```
 
-To work around this issue, it is recommended to run your tests with the `--pool=forks` flag.
+## ETag Support
 
-See [#169](https://github.com/flipt-io/flipt-client-sdks/issues/169) for more information.
+The default fetcher uses [ETag HTTP headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag) to reduce the number of requests to the Flipt instance. This is useful in scenarios where the flag state is not frequently updated and you want to reduce the load on the server.
```

Comment:
Small clarification: doesn't change the number of requests (just avoids resending previously observed state):
```suggestion
The default fetcher uses [ETag HTTP headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag) to reduce overhead building and sending previously observed snapshots. This is useful in scenarios where the flag state is not frequently updated and you want to reduce the load on the server.
```